recipes/exiv2/all/conanfile.py

```diff
@@ -67,56 +52,57 @@ def configure(self):
             self.provides.append("xmp-toolkit-sdk")
 
     def requirements(self):
-        self.requires("libiconv/1.16")
+        self.requires("libiconv/1.17")
         if self.options.with_png:
-            self.requires("libpng/1.6.37")
+            self.requires("libpng/1.6.38")
         if self.options.with_xmp == "bundled":
-            self.requires("expat/2.4.3")
+            self.requires("expat/2.4.9")
         if self.options.with_curl:
-            self.requires("libcurl/7.80.0")
+            self.requires("libcurl/7.85.0")
 
     def validate(self):
         if self.options.with_xmp == "external":
             raise ConanInvalidConfiguration("adobe-xmp-toolkit is not available on cci (yet)")
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["EXIV2_BUILD_SAMPLES"] = False
+        tc.variables["EXIV2_BUILD_EXIV2_COMMAND"] = False
+        tc.variables["EXIV2_ENABLE_PNG"] = self.options.with_png
+        tc.variables["EXIV2_ENABLE_XMP"] = self.options.with_xmp == "bundled"
+        tc.variables["EXIV2_ENABLE_EXTERNAL_XMP"] = self.options.with_xmp == "external"
+        # NLS is used only for tool which is not built
+        tc.variables["EXIV2_ENABLE_NLS"] = False
+        tc.variables["EXIV2_ENABLE_WEBREADY"] = self.options.with_curl
+        tc.variables["EXIV2_ENABLE_CURL"] = self.options.with_curl
+        tc.variables["EXIV2_ENABLE_SSH"] = False
+
+        if is_msvc(self):
+            tc.variables["EXIV2_ENABLE_DYNAMIC_RUNTIME"] = "MD" in msvc_runtime_flag(self)
+        # set PIC manually because of object target exiv2_int
+        tc.variables["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
```

Comment:
```suggestion
        tc.cache_variables["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
```
For those cached variables in the project, prefer `cache_variables`: https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html#cache-variables